lib/model/query/forms.js

```diff
@@ -674,25 +699,31 @@ inner join
 where ${sqlEquals(condition)} and forms."deletedAt" is ${deleted ? sql`not` : sql``} null
 ${(actorId == null) ? sql`` : sql`and (form_defs."publishedAt" is not null or filtered."showDraft" = 1)`}
 ${(actorId == null) ? sql`` : sql`and (state != 'closed' or filtered."showNonOpen" = 1)`}
-order by coalesce(form_defs.name, "xmlFormId") asc`;
+ORDER BY ${version === Form.WithoutDef ? sql`"xmlFormId"` : sql`coalesce(form_defs.name, "xmlFormId")`} ASC`;
 });
+/* eslint-enable indent */
+
+const _get = (exec, options, version, xml, deleted, actorId) => {
+  if (xml !== Form.IncludeXml && xml !== Form.WithoutXml) throw new Error('Must use Form.WithoutXml or Form.IncludeXml; actually got: ' + xml.toString());
+  if (version === Form.WithoutDef && xml === Form.IncludeXml) throw new Error('Cannot request xml without a form def');
 
-const _getWithoutXml = extender(Form, Form.Def)(Form.Extended, Actor.into('createdBy'))(_getSql);
-const _getWithXml = extender(Form, Form.Def, Form.Xml)(Form.Extended, Actor.into('createdBy'))(_getSql);
-const _get = (exec, options, xml, version, deleted, actorId) =>
-  ((xml === true) ? _getWithXml : _getWithoutXml)(exec, options, version, deleted, actorId);
+  const frames = [ Form ];
+  if (version !== Form.WithoutDef) frames.push(Form.Def);
```

Comment:
I was confused about why `Form.WithoutDef` would be passed through the `version` parameter, but after thinking about it for a long time, I've come up with this reason: If you don't need the def for the form, then it really doesn't matter if you're fetching a published or unpublished version, or a specific version...

That's why `Form.WithoutDef` is an ok thing to pass in the version string especially for the resources about listing form versions or deleting forms.

Reply:
Ooh, yes that would help a bit.